src/Simplify.elm

```diff
@@ -2872,6 +2873,49 @@ stringReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
                 checkInfo
 
 
+stringSliceChecks : CheckInfo -> List (Error {})
+stringSliceChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg ) of
+        ( Node _ (Expression.Integer 0), _ ) ->
```

Comment:
I think the order of checks could be improved, so that `String.slice 0 0 str` does not end up being `String.left 0 str`. You are adding support for this in this PR, but by changing the order we could avoid an additional round-trip.